src/Tasks.UnitTests/RARPrecomputedCache_Tests.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using Microsoft.Build.Framework;
+using Microsoft.Build.UnitTests;
+using Microsoft.Build.Utilities;
+using Shouldly;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Xunit;
+
+namespace Microsoft.Build.Tasks.UnitTests
+{
+    public class RARPrecomputedCache_Tests
+    {
+        private Dictionary<string, Guid> guidStore = new Dictionary<string, Guid>();
+
+        private Guid calculateMvid(string path)
+        {
+            if (!guidStore.ContainsKey(path))
+            {
+                guidStore.Add(path, Guid.NewGuid());
+            }
+            return guidStore[path];
+        }
+
+        [Fact]
+        public void TestPrecomputedCacheOutput()
+        {
+            using (TestEnvironment env = TestEnvironment.Create())
+            {
+                TransientTestFile standardCache = env.CreateFile(".cache");
+                ResolveAssemblyReference t = new ResolveAssemblyReference();
+                t._cache = new SystemState();
+                t._cache.instanceLocalFileStateCache = new Dictionary<string, SystemState.FileState>() {
+                    { Path.Combine(standardCache.Path, "assembly1"), new SystemState.FileState(DateTime.Now) },
+                    { Path.Combine(standardCache.Path, "assembly2"), new SystemState.FileState(DateTime.Now) { Assembly = new Shared.AssemblyNameExtension("hi") } } };
+                t._cache.IsDirty = true;
+                t.StateFile = standardCache.Path;
+                t.WriteStateFile(calculateMvid);
+                int standardLen = File.ReadAllText(standardCache.Path).Length;
+                File.Delete(standardCache.Path);
+                standardLen.ShouldBeGreaterThan(0);
```

Comment:
Is this a strong enough check? Should we directly compare `File.ReadAllText` to exactly what we think the output should be?

Reply:
I'd love to, but unfortunately, that output would differ depending on serialization method.

Reply:
I don't think I understand the purpose of this test. Isn't it a subset of the other one that does a round trip?

Reply:
Kinda, yeah. I still favor leaving it in because 1) it's already written, and 2) if there's a problem here (nontrivial possibility, since this is new), having a simpler test will more readily provide insight as to what exactly is broken.